dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,345 @@
+*# Treasury add transactions - method to credit the treasury account.
+*# Treasury spend transactions - method to debit the treasury account.
+
+==Motivation==
+
+At this present time all Decred treasury payouts are handled by DHG. This is
+undesirable for various reasons, such as
+
+* sovereignty over treasury spends is not vested in the stakeholders
+* risk of theft of funds
+* risk of key project members abandoning the project
+* risk of coercion being used against key project members
+* process of treasury spending is insufficiently transparent
+
+The proposed solution will semi-automate the release of treasury funds while
+diminishing key project members' influence over the treasury spends.
+
+==Process==
+
+Currently, every block that is mined must include a coinbase transaction that
+that generates a utxo of a specific amount that requires a 2-of-3 multisig to
+spend. This is enforced by consensus. During payout time, DHG creates and signs
+transactions to all payees. This process is not adequate to support the
+features proposed in this DCP.
+
+Essentially, the treasury is converted from a utxo based model to an account
+based model. The account balance is determined by an initial state followed by
+state changes that are recorded on the blockchain. This process "burns" and
+"creates" utxos as needed. Sending utxos to the treasury burns the coins and
+adds value to the treasury account and sending from the treasury creates utxos.
+This process is analogous to the ticket purchases and redeems. The treasury
+changes shall be added to the staking mechanism.
+
+This DCP introduces three distinct new processes.
+# In order to credit the treasury account every block the coinbase transaction is split into two transactions. In the stake tree the first transaction is now a treasurybase. Treasurybase is a newly introduced term that is defined as the transaction that credits the treasury account on every block. The value of this transaction is the same (exception will be specified later in this document) as it was in the coinbase transaction. The coinbase transaction remains identical except for the noted change.
+# Users need to be able to send utxos to the treasury account. This is a coin burn that is analogous to purchasing a ticket. This feature is also going to be used to move currently unused funds in the treasury to the new treasury account.
+# Spending treasury funds now requires stakeholders to vote on the expenditure. The Pi key operators creates a "treasury spend" transaction and signs it. This transaction is then broadcast onto the network and it enters the mempool. Voting wallets will detect this transaction and they will automatically vote Yes/No on the expenditure based on the embedded key and signature (if enabled). Treasury expenditure votes are collected during the consensus voting process. If the treasury spend transaction collects enough yes votes it is included in the chain and payees will receive their funds.
```

Comment:
Again, this is a specification, so it should call out the expected and required behavior, not discuss what `dcrwallet` will do.

Also, it's unclear to me what `based on the embedded key and signature (if enabled).` it intended to mean.

Perhaps the following:

```suggestion
# Spending treasury funds now requires stakeholders to vote on the expenditure. The Pi key operators create a <code>treasury spend</code> transaction and sign it.  This transaction is then broadcast onto the network where it enters the <code>mempool</code>.  Voting wallets are expected to detect this transaction and vote <code>Yes</code> or <code>No</code> on the expenditure based on stakeholder preference via the existing <code>vote</code> stake transactions.  The <code>treasury spend</code> transaction is not valid until enough <code>Yes</code> votes are cast by the stakeholders at which point it is included in a block thereby finalizing the payouts.
```